ebmbot/slack.py

```diff
@@ -45,11 +47,42 @@ def notify_slack(
             attempts += 1
             sleep(1)
             error = err
+    # The message has failed to post.
+    # Modify the Slack message to report the error and try posting it to the channel
+    COULD_NOT_NOTIFY_SLACK = "Could not notify slack"
+    try:
+        msg_kwargs["text"] = COULD_NOT_NOTIFY_SLACK
+        msg_kwargs["blocks"] = get_slack_error_blocks(
+            COULD_NOT_NOTIFY_SLACK, message_text, error=error
+        )
+        slack_client.chat_postMessage(**msg_kwargs)
+    except Exception:
+        # Not even the error message could not be posted to Slack, so
+        pass
 
-    logger.error(
-        "Could not notify slack",
-        channel=channel,
-        message=message_text,
-        thread_ts=thread_ts,
-        error=error,
+    finally:
+        # Log the error
+        logger.error(
+            COULD_NOT_NOTIFY_SLACK,
+            channel=channel,
+            message=message_text,
+            thread_ts=thread_ts,
+            error=error,
+        )
+
+
+def get_slack_error_blocks(header_text, message_text, error):
+    error_text = str(error)
+    if len(error_text) > 2994:
+        error_text = error_text[:2991] + "..."
+    if len(message_text) > 2994:
+        message_text = message_text[:2991] + "..."
```

Comment:
Also a tiny nit: you could avoid the repetition with a tiny helper function, e.g.
```
def truncate_text_for_block(text):
    if len(text) > 2994:
        return text[:2991] + "..."
```
And then you can plug that straight into the fstring
```
...
f"```{truncate_text_for_block(str(error_text))}```"
```

Reply:
I don't think this is covered by a test anywhere

Reply:
Made a test_utils.py file to test this